wagtail/models/__init__.py

```diff
@@ -2172,21 +2216,41 @@ class PageRevision(models.Model):
     )
 
     objects = models.Manager()
+    page_revisions = PageRevisionsManager()
     submitted_revisions = SubmittedRevisionsManager()
 
+    @cached_property
+    def content_object(self):
+        return self.base_content_type.get_object_for_this_type(pk=self.object_id)
+
+    @cached_property
```

Comment:
I don't think we need this property to be cached as `content_object` is already cached